tiantianXia/assignments/treesDFS/lc112/lc112.js

```diff
@@ -11,10 +11,49 @@ class TreeNode {
   }
 };
 
+/**
+ * inputs: 
+ * @param {*} root: the root node of the binary tree 
+ * @param {*} sum: the  sum that compare with all the paths
+ * 
+ * outputs: 
+ * @returns boolean value that if one path is equal to the sum
+ * 
+ * step 1: using recursion to traverse all the tree nodes
+ * step 2: add up the current sum 
+ * step 3: if reach the leaf node compare the current sum and sum
+ * step 4: if not the leaf pass the current sum to left or right child
+ * 
+ * variable currentSum: add up the current number to the currentSum
+ * 
+ * using recursion function to traverse all the tree nodes
+ *    if the node is null value return false
+ *    add up current value to the current sum
+ *    if find the leaf node that no left and right child
+ *        compare sum and current sum and return the value
+ *    if the left or right child returns a true value the return ture
+ *    else return false
+ * end function
+ * return the result of the recursion function
+ * 
+ * edge cases: if the  root is null => false;
```

Comment:
Great approach here! :100: Its more like psuedocode but from it I know what you are trying to accomplish.